### ISSUE TITLE
PanicException when running `unique()` on object/null dtype

### ISSUE THREAD

**Issue by guyrosin** [pola-rs/polars#10670]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
df = pl.DataFrame({"a": pl.lit(None)})
df.unique()
```

Result:
```python
thread '<unnamed>' panicked at 'should be hashable: PyErr { type: <class 'TypeError'>, value: TypeError("unhashable type: 'Expr'"), traceback: None }', src/conversion.rs:934:68
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
[/tmp/ipykernel_6583/1660334235.py](https://vscode-remote+ssh-002dremote-002b7b22686f73744e616d65223a224750552d415753227d.vscode-resource.vscode-cdn.net/tmp/ipykernel_6583/1660334235.py) in ()
----> 1 df.unique()

[~/playground/.venv/lib/python3.9/site-packages/polars/dataframe/frame.py](https://vscode-remote+ssh-002dremote-002b7b22686f73744e616d65223a224750552d415753227d.vscode-resource.vscode-cdn.net/home/ec2-user/playground/notebooks/~/playground/.venv/lib/python3.9/site-packages/polars/dataframe/frame.py) in unique(self, subset, keep, maintain_order)
   8289         """
   8290         return (
-> 8291             self.lazy()
   8292             .unique(subset=subset, keep=keep, maintain_order=maintain_order)
   8293             .collect(no_optimization=True)

[~/playground/.venv/lib/python3.9/site-packages/polars/utils/deprecation.py](https://vscode-remote+ssh-002dremote-002b7b22686f73744e616d65223a224750552d415753227d.vscode-resource.vscode-cdn.net/home/ec2-user/playground/notebooks/~/playground/.venv/lib/python3.9/site-packages/polars/utils/deprecation.py) in wrapper(*args, **kwargs)
     91                 old_name, new_name, kwargs, function.__name__, version
     92             )
---> 93             return function(*args, **kwargs)
     94 
     95         return wrapper

[~/playground/.venv/lib/python3.9/site-packages/polars/lazyframe/frame.py](https://vscode-remote+ssh-002dremote-002b7b22686f73744e616d65223a224750552d415753227d.vscode-resource.vscode-cdn.net/home/ec2-user/playground/notebooks/~/playground/.venv/lib/python3.9/site-packages/polars/lazyframe/frame.py) in collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, no_optimization, slice_pushdown, comm_subplan_elim, comm_subexpr_elim, streaming)
   1693             streaming,
   1694         )
-> 1695         return wrap_df(ldf.collect())
   1696 
   1697     def sink_parquet(

PanicException: should be hashable: PyErr { type: , value: TypeError("unhashable type: 'Expr'"), traceback: None }
```

df looks like this:
```python
shape: (1, 1)
┌────────┐
│ a      │
│ ---    │
│ object │
╞════════╡
│ null   │
└────────┘
```

The above exception is raised for any literal I tried (e.g., `pl.DataFrame({"a": pl.lit(1)})`)

### Issue description

`unique()` seems to fail to run on a Series or DataFrame of dtype `object` or `null`.

### Expected behavior

I expected `unique()` to return a Series containing `null`, similarly to what happens when we have a Series with the same values but a different dtype:

```python
df = pl.DataFrame({"a": [None]}).unique()
df = df.unique()
print(df)
```

```python
shape: (1, 1)
┌──────┐
│ a    │
│ ---  │
│ f32  │
╞══════╡
│ null │
└──────┘
```

### Installed versions

<details>

```
--------Version info---------
Polars:              0.18.15
Index type:          UInt32
Platform:            Linux-4.14.252-195.483.amzn2.x86_64-x86_64-with-glibc2.26
Python:              3.9.16 (main, Apr 27 2023, 12:45:53) 
[GCC 7.3.1 20180712 (Red Hat 7.3.1-15)]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         2.2.1
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              2023.6.0
matplotlib:          3.7.2
numpy:               1.24.4
pandas:              2.0.3
pyarrow:             12.0.1
pydantic:            <not installed>
sqlalchemy:          2.0.20
xlsx2csv:            <not installed>
xlsxwriter:          3.1.2```

</details>

**Comment by ritchie46** [comment]
On objects we will not support unique. For nulls we should fix this.

**Comment by guyrosin** [comment]
Fixed by #13307
